external_demo_annotator/demo_annotator/vep_annotator.py

```diff
@@ -0,0 +1,398 @@
+                    annotatable.pos,
+                    annotatable.end_position,
+                    f"{annotatable.ref}/{annotatable.alt}",
+                    "+",
+                    idx,
+                ])
+        file.flush()
+
+    def read_output(
+        self, file: TextIO, contexts: list[dict[str, Any]],
+    ) -> None:
+        """Read and return subprocess output contents."""
+        for context in contexts:
+            context["gene"] = []
+            context["gene_id"] = []
+            context["feature"] = []
+            context["feature_type"] = []
+            context["consequence"] = []
+            context["location"] = []
+            context["allele"] = []
+            context["cdna_position"] = []
+            context["cds_position"] = []
+            context["protein_position"] = []
+            context["amino_acids"] = []
+            context["codons"] = []
+            context["existing_variation"] = []
+            context["impact"] = []
+            context["distance"] = []
+            context["strand"] = []
+            context["symbol_source"] = []
+
+        reader = csv.reader(
+            filter(lambda row: row[0] != "#", file),
+            delimiter="\t",
+        )
+
+        for row in reader:
+            idx = int(row[0]) - 1
+            context = contexts[idx]
+            context["allele"].append(row[1])
+            context["location"].append(row[2])
+            context["gene_id"].append(row[3])
+            context["feature"].append(row[4])
+            context["feature_type"].append(row[5])
+            context["consequence"].append(row[6])
+            context["cdna_position"].append(row[7])
+            context["cds_position"].append(row[8])
+            context["protein_position"].append(row[9])
+            context["amino_acids"].append(row[10])
+            context["codons"].append(row[11])
+            context["existing_variation"].append(row[12])
+            context["impact"].append(row[13])
+            context["distance"].append(row[14])
+            context["strand"].append(row[15])
+            context["gene"].append(row[17])
+            context["symbol_source"].append(row[18])
+
+        for context in contexts:
+            gene_consequences = []
+            for gene, consequence in zip(
+                context["gene"], context["consequence"], strict=True,
+            ):
+                gene_consequences.append(f"{gene}:{consequence}")
+            context["gene_consequence"] = gene_consequences
+
+            the_consequences: list[str] = []
+            for conseq in context["consequence"]:
+                the_consequences.extend(conseq.split(","))
+
+            context["worst_consequence"] = [sorted(
+                the_consequences, key=lambda x: CONSEQUENCES[x],
+            )[0]]
+            context["highest_impact"] = [sorted(
+                context["impact"], key=lambda x: IMPACTS[x],
+            )[0]]
+
+
+class DemoVEPCacheAnnotator(VEPAnnotatorBase):
```

Comment:
We should remove `Demo` from the name of this annotator. Please rename to `VEPCacheAnnotator`